packages/mcdoc/src/binder/index.ts

```diff
@@ -113,6 +114,7 @@ namespace ModuleSymbolData {
 
 export interface TypeDefSymbolData {
 	typeDef: McdocType
+	simplifiedTypeDef?: SimplifiedMcdocType
 }
```

Comment:
Currently I am just amending the symbol, which means SimplifiedMcdocType is spreading again.
I think this should have the advantage that the cached value is removed for free when the symbol changes.

However, I am not sure how good this approach is. I didn't really figure out another good way to implement this though.

Reply:
This looks really dangerous to me, especially because `simplify` is not pure anymore and because simplified types also depend on runtime values.

Reply:
> especially because `simplify` is not pure anymore

If you mean the struct field thing from last PR, this is gone again in this PR, as I made field deduplication unnessary.

If you mean sth else, in what way is it not pure anymore?

> because simplified types also depend on runtime values.

The only part where runtime types are actually needed is for resolving dynamic dispatcher / indexed type indices. In these cases, I do not cache anything (dynamicData: true is returned)